### PR TITLE
Don't select more inputs than allowed

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Don't select more inputs than allowed, also check max outputs

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes #issue_number`.

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Added tests

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/api/block_builder/input_selection/core/requirement/amount.rs [hunk 1]
```diff
@@ -354,28 +409,16 @@ impl InputSelection {
                 );
                 log::debug!("Triggering another amount round as non-basic outputs need to be transitioned first");
 
-                self.available_inputs
-                    .retain(|input| !amount_selection.newly_selected_inputs.contains_key(input.output_id()));
+                if self.selected_inputs.len() + amount_selection.newly_selected_inputs.len() <= INPUT_COUNT_MAX.into() {
```
Comment (thibault-martinez) [inline client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Isn't there a risk of infinite loop if this is not the case?

Comment (Thoralf-M) [reply client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Actually there is no point in having this loop at all? :carlos:

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Little conflict, sorry

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]
lgtm!

**Pull request merged**